File: frontend/src/interfaces/problem_report.ts
import type { ProblemAttachment } from "./problem_attachment";
import type { ProblemReply } from "../interfaces/problem_reply";
import type { User } from "./User";
import type { Game } from "./Game";

/**
 * รูปแบบ ProblemReport ตาม backend (gorm.Model)
 * - ID: number (พิมพ์ใหญ่เพราะ GORM คืนมาเป็น "ID")
 * - field อื่นเป็น lower_case ตาม json
 */
export interface ProblemReport {
  ID: number;             // primary key

  title: string;
  description: string;
  status: string;         // "pending" | "resolved"

  created_at?: string;
  resolved_at?: string;

  // foreign keys
  user_id: number;        // ✅ ใช้ตัวนี้ตอน createNotification
  game_id: number;

  // preload relations
  user?: User;
  game?: Game;

  // admin reply
  reply?: string; // latest reply message

  // attachments
  attachments?: ProblemAttachment[];

  // all replies from admin
  replies?: ProblemReply[];
}

File: backend/entity/problem_report.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

// คำร้องจากลูกค้า
type ProblemReport struct {
	gorm.Model

	Title       string `json:"title"       gorm:"type:varchar(200);not null"`
	Description string `json:"description" gorm:"type:text;not null"`
	Category    string `json:"category"    gorm:"type:varchar(100);default:'Other';index"`

	// ผู้ส่งคำร้อง
	UserID uint  `json:"user_id" gorm:"not null;index"`
	User   *User `json:"user"    gorm:"foreignKey:UserID"`

	// เกมที่เกี่ยวข้อง (ถ้ามี)
	GameID uint  `json:"game_id" gorm:"index"`
	Game   *Game `json:"game"    gorm:"foreignKey:GameID"`

	// สถานะการดำเนินงาน
	// ใช้ค่า: open | in_progress | resolved
	Status     string     `json:"status"      gorm:"type:varchar(30);default:'open';index"`
	ResolvedAt *time.Time `json:"resolved_at"` // อนุญาตให้เป็น null

	// แนบไฟล์ของลูกค้า
	Attachments []ProblemAttachment `json:"attachments" gorm:"foreignKey:ReportID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`

	// ความสัมพันธ์กับตารางตอบกลับของแอดมิน
	Replies []ProblemReply `json:"replies" gorm:"foreignKey:ReportID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}
